src/phoenix/experimental/evals/functions/classify.py

```diff
@@ -166,13 +170,20 @@ async def consumer(
                     continue
                 else:
                     tqdm.write("Worker timeout, requeuing")
-                    await queue.put(item)
-            except Exception:
-                tqdm.write(f"Exception in worker: {traceback.format_exc()}")
-                if self.exit_on_error:
-                    self._TERMINATE.set()
+                    await queue.put((self.requeue_priority, item))
+            except Exception as exc:
+                if retry_count := abs(priority) <= self.max_retries:
```

Comment:
```suggestion
                if (retry_count := abs(priority)) <= self.max_retries:
```

I'm not sure of where the walrus operator falls in the order of operations, but this makes it more explicit.